### PR TITLE
sc-6127 Enhance poipath for insertion to generate templates dynamically

### PR DESCRIPTION
**Pull request by alexpartsch**
By extending poipath for insertion, it could be made possible to generate templates for generator tests dynamically.

https://app.shortcut.com/ddsg/story/6127/enhance-poipath-for-insertion-to-generate-templates-dynamically

### CODE REVIEW

File: src/main/java/com/docutools/poipath/xssf/XSSFBuilder.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+   *
+   * @param content the cell value
+   * @param clazz cell value Java type
+   * @param <V> cell value java type
+   * @return {@code this}
+   */
+  public <V> XSSFBuilder<T> value(V content, Class<V> clazz) {
+    if (bodyElement instanceof XSSFCell cell) {
+      try {
+        var method = XSSFCell.class.getMethod("setCellValue", clazz);
```
Comment (AntonOellerer) [inline src/main/java/com/docutools/poipath/xssf/XSSFBuilder.java:113]: Wouldnt `cell.setCellValue` work here as well?

File: src/main/java/com/docutools/poipath/xssf/XSSFWorkbookWrapper.java [hunk 2]
```diff
@@ -21,4 +21,8 @@ public int numberOfSheets() {
     return workbook.getNumberOfSheets();
   }
 
+  public static XSSFWorkbookWrapper wrap(XSSFWorkbook workbook) {
```
Comment (AntonOellerer) [inline src/main/java/com/docutools/poipath/xssf/XSSFWorkbookWrapper.java:24]: why do we need this if we have the constructor?

File: src/main/java/com/docutools/poipath/xwpf/XWPFBuilder.java [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  }
+
+  /**
+   * Creates a new {@link XWPFParagraph} with the given text
+   *
+   * @param text paragraph content
+   * @return {@code this}
+   */
+  public XWPFBuilder<T> paragraph(String text) {
+    var paragraphTexts = text.split("\n");
```
Comment (AntonOellerer) [inline src/main/java/com/docutools/poipath/xwpf/XWPFBuilder.java]: Do we really need to create one paragraph per line?

File: src/test/java/com/docutools/poipath/xwpf/XWPFTests.java [hunk 4]
```diff
@@ -46,7 +46,7 @@ void findTables() throws IOException {
   @Test
   void findPictures() throws IOException {
     try (var document = Documents.resource("/XWPF/pictureTest.docx")) {
-      var picture = new XWPFDocumentWrapper(document).bodyElement(0).asParagraph().run(0).pictures();
+      var picture = Objects.requireNonNull(new XWPFDocumentWrapper(document).bodyElement(0).asParagraph()).run(0).pictures();
```
Comment (AntonOellerer) [inline src/test/java/com/docutools/poipath/xwpf/XWPFTests.java:49]: Why is this necessary?

File: src/main/java/com/docutools/poipath/xssf/XSSFBuilder.java (hunk 1 above)
Comment (alexpartsch) [reply src/main/java/com/docutools/poipath/xssf/XSSFBuilder.java:113]: Nope, we have multiple overloads of `setCellValue` and due to the generic value the compiler can't figure that out.

File: src/main/java/com/docutools/poipath/xwpf/XWPFBuilder.java (hunk 3 above)
Comment (alexpartsch) [reply src/main/java/com/docutools/poipath/xwpf/XWPFBuilder.java]: Nope, fixed.

File: src/main/java/com/docutools/poipath/xssf/XSSFWorkbookWrapper.java (hunk 2 above)
Comment (alexpartsch) [reply src/main/java/com/docutools/poipath/xssf/XSSFWorkbookWrapper.java:24]: Removed.

File: src/test/java/com/docutools/poipath/xwpf/XWPFTests.java (hunk 4 above)
Comment (alexpartsch) [reply src/test/java/com/docutools/poipath/xwpf/XWPFTests.java:49]: Since `asParagraph` can return `null`.
